### PR TITLE
test: add consensus to e2e flow test

### PR DESCRIPTION
**Pull request by dan-starkware**

### CODE REVIEW

**Comment by dan-starkware** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1811?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1811** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1811?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1810** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1809** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1809?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1808** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1808?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1807** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1807?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1806** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1806?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1774** <a href="https://app.graphite.dev/github/pr/starkware-libs/sequencer/1774?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dan-starkware and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Itay-Tsabary-Starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: 0 of 7 files reviewed, 4 unresolved discussions (waiting on @dan-starkware)

___
*[`crates/tests-integration/src/integration_test_utils.rs` line 72 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvH_FmCBhKhLO2XHMx:-OAvH_FmCBhKhLO2XHMy:b-yvn0no) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/src/integration_test_utils.rs#L72)):*
> ```XML
> }
> 
> fn create_consensus_manager_config()
> ```

Current name is confusing, as there are other `create_X_config` that just return a config object, while this variant also returns the channels. Could you please rename?



_Code quote:_
```XML
create_consensus_manager_config
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 44 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvH2cG2r7NYpbwE0d5:-OAvH2cG2r7NYpbwE0d6:bn4qaq3) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/tests/end_to_end_test.rs#L44)):*
> ```XML
>         .expect("listen to broadcasted messages should finish in time");
>     });
>     join_handle.await.expect("Task should succeed");
> ```

Please wrap the "await for something to happen within a timeout" as a utility function (or add a TODO, can include my name as well).



_Code quote:_
```XML
    let join_handle = tokio::spawn(async move {
        tokio::time::timeout(
            LISTEN_TO_BROADCAST_MESSAGES_TIMEOUT,
            listen_to_broadcasted_messages(
                mock_running_system.consensus_proposals_channels,
                &expected_batched_tx_hashes,
            ),
        )
        .await
        .expect("listen to broadcasted messages should finish in time");
    });
    join_handle.await.expect("Task should succeed");
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 51 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvI-tHBpZE7mD6t28M:-OAvI-tHBpZE7mD6t28N:bb752ax) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/tests/end_to_end_test.rs#L51)):*
> ```XML
>     expected_batched_tx_hashes: &[TransactionHash],
> ) {
>     // TODO(Dan, Guy): retrieve chian ID (from the config?).
> ```

Typo



_Code quote:_
```XML
chian
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 51 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvIBAFCUe-wQ9jfBfc:-OAvIBAFCUe-wQ9jfBfd:b984vpb) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/tests/end_to_end_test.rs#L51)):*
> ```XML
>     expected_batched_tx_hashes: &[TransactionHash],
> ) {
>     // TODO(Dan, Guy): retrieve chian ID (from the config?).
> ```

I'd suggest modifying `IntegrationTestSetup` to hold it as a member, and instantiate the value using `StorageTestSetup`.



_Code quote:_
```XML
(from the config?)
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 71 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvJLkD9FdH4yJ6oiHq:-OAvJLkD9FdH4yJ6oiHr:b-m9ab2) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/tests/end_to_end_test.rs#L71)):*
> ```XML
>             );
>         }
>     }
> ```

I didn't review this part, I suggest asking someone more familiar with the consensus to do so. If there's no on else I'll dig into this though, please lmk.



_Code quote:_
```XML
    let mut broadcasted_messages_receiver =
        consensus_proposals_channels.broadcasted_messages_receiver;
    let mut received_tx_hashes = vec![];
    while received_tx_hashes.len() < expected_batched_tx_hashes.len() {
        let (message, _broadcasted_message_metadata) = broadcasted_messages_receiver
            .next()
            .await
            .unwrap_or_else(|| panic!("Expected to receive a message from the broadcast topic"));
        if let ProposalPart::Transactions(transactions) = message.unwrap() {
            received_tx_hashes.append(
                &mut transactions
                    .transactions
                    .iter()
                    .map(|tx| tx.calculate_transaction_hash(&chain_id).unwrap())
                    .collect(),
            );
        }
    }
```

**Review by dan-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: 0 of 7 files reviewed, 3 unresolved discussions (waiting on @Itay-Tsabary-Starkware)

___
*[`crates/tests-integration/src/integration_test_utils.rs` line 72 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvH_FmCBhKhLO2XHMx:-OAvKjuoE9ejlNqv1KyB:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/src/integration_test_utils.rs#L72)):*
<details><summary><i>Previously, Itay-Tsabary-Starkware wrote…</i></summary><blockquote>

Current name is confusing, as there are other `create_X_config` that just return a config object, while this variant also returns the channels. Could you please rename?
</blockquote></details>

Done.
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 44 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvH2cG2r7NYpbwE0d5:-OAvLUFD67yBsmNujErE:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/tests/end_to_end_test.rs#L44)):*
<details><summary><i>Previously, Itay-Tsabary-Starkware wrote…</i></summary><blockquote>

Please wrap the "await for something to happen within a timeout" as a utility function (or add a TODO, can include my name as well).
</blockquote></details>

Done.
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 51 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OAvIBAFCUe-wQ9jfBfc:-OAvL6maA56WuA8AgC14:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/cb12e5bc155b9f5cad3221a677ef549348b4afe9/crates/tests-integration/tests/end_to_end_test.rs#L51)):*
<details><summary><i>Previously, Itay-Tsabary-Starkware wrote…</i></summary><blockquote>

I'd suggest modifying `IntegrationTestSetup` to hold it as a member, and instantiate the value using `StorageTestSetup`.
</blockquote></details>

Done.

**Review by Itay-Tsabary-Starkware — APPROVED** [review]
:lgtm:

> Reviewed 5 of 7 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-:-OAv_7nJABIIt9IWtiBb:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dan-starkware)

**Review by guy-starkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: 5 of 7 files reviewed, 1 unresolved discussion (waiting on @dan-starkware and @Itay-Tsabary-Starkware)

___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 63 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB53FKh7BTjahBTb-nW:-OB53FKi1rsoLFpvDLsT:blqn41) ([raw file](https://github.com/starkware-libs/sequencer/blob/6b7dc98b73d7c82b600334418130f68cfef0f03f/crates/tests-integration/tests/end_to_end_test.rs#L63)):*
> ```XML
>             .next()
>             .await
>             .unwrap_or_else(|| panic!("Expected to receive a message from the broadcast topic"));
> ```

Is there a reason to use `unwrap_or_else(panic!(...))` instead of just `expect(...)`?

**Review by matan-starkware — CHANGES_REQUESTED** [review]
> Reviewed 3 of 7 files at r1, 2 of 2 files at r2, 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 8 unresolved discussions (waiting on @dan-starkware and @Itay-Tsabary-Starkware)

___
*[`crates/batcher_types/src/batcher_types.rs` line 61 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB4b4Sh3BjKo_-XY__O:-OB4b4Sh3BjKo_-XY__P:b-h9ow7t) ([raw file](https://github.com/starkware-libs/sequencer/blob/6b7dc98b73d7c82b600334418130f68cfef0f03f/crates/batcher_types/src/batcher_types.rs#L61)):*
> ```XML
> 
> #[derive(Clone, Debug, Serialize, Deserialize)]
> // Same as propose?
> ```

I prefer phrasing this as a TODO.



_Suggestion:_
```XML
// TODO(Dan): Consider unifying with BuildProposalInput as they have the same fields.
```
___
*[`crates/tests-integration/Cargo.toml` line 49 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB4bQJS7CTz4fg1P7VN:-OB4bQJS7CTz4fg1P7VO:bhs0m4a) ([raw file](https://github.com/starkware-libs/sequencer/blob/6b7dc98b73d7c82b600334418130f68cfef0f03f/crates/tests-integration/Cargo.toml#L49)):*
> ```TOML
> futures.workspace = true
> papyrus_network.workspace = true
> papyrus_protobuf.workspace = true
> ```

Why repeat this from the regular set of dependencies?



_Code quote:_
```TOML
papyrus_protobuf.workspace = true
```
___
*[`crates/tests-integration/src/integration_test_utils.rs` line 83 at r3](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB4cKBF2O9MQXsBONwE:-OB4cKBG3ky1AK_rUBTJ:bsawvww) ([raw file](https://github.com/starkware-libs/sequencer/blob/6b7dc98b73d7c82b600334418130f68cfef0f03f/crates/tests-integration/src/integration_test_utils.rs#L83)):*
> ```XML
>         consensus_config: ConsensusConfig {
>             start_height: BlockNumber(1),
>             consensus_delay: Duration::from_secs(3),
> ```

How did you decide on this? The goal is to make this as small as possible since it's just extra latency on the test's startup.



_Code quote:_
```XML
            consensus_delay: Duration::from_secs(3),
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 45 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB5jNFj7Xqjkmeh13mF:-OB5jNFj7Xqjkmeh13mG:box87zz) ([raw file](https://github.com/starkware-libs/sequencer/blob/35b4aa651a8ad7c1cfde431fa0e071e5c1a552b6/crates/tests-integration/tests/end_to_end_test.rs#L45)):*
> ```XML
>         .expect("listen to broadcasted messages should finish in time");
>     });
>     join_handle.await.expect("Task should succeed");
> ```

Why bother spawning if you just await immediately?



_Suggestion:_
```XML
    tokio::time::timeout(
        LISTEN_TO_BROADCAST_MESSAGES_TIMEOUT,
        listen_to_broadcasted_messages(
            mock_running_system.consensus_proposals_channels,
            &expected_batched_tx_hashes,
        ),
    )
    .await
    .expect("listen to broadcasted messages should finish in time");
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 48 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB5his52F7bX09lOF9D:-OB5his65JBnC_1r27bW:b-sg530) ([raw file](https://github.com/starkware-libs/sequencer/blob/35b4aa651a8ad7c1cfde431fa0e071e5c1a552b6/crates/tests-integration/tests/end_to_end_test.rs#L48)):*
> ```XML
> }
> 
> async fn listen_to_broadcasted_messages(
> ```

I thought our style was to define everything that is not a test above all the tests?



_Code quote:_
```XML
async fn listen_to_broadcasted_messages(
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 71 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB5lQmO5sg7VmGISc-b:-OB5lQmO5sg7VmGISc-c:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/35b4aa651a8ad7c1cfde431fa0e071e5c1a552b6/crates/tests-integration/tests/end_to_end_test.rs#L71)):*
> ```XML
>                     .map(|tx| tx.calculate_transaction_hash(&chain_id).unwrap())
>                     .collect(),
>             );
> ```





_Suggestion:_
```XML
            received_tx_hashes.extend(
                transactions
                    .transactions
                    .into_iter()
                    .map(|tx| tx.calculate_transaction_hash(&chain_id).unwrap()),
            );
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 74 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OB5lilb8Sxz9CNkVgEj:-OB5lilb8Sxz9CNkVgEk:b42iijd) ([raw file](https://github.com/starkware-libs/sequencer/blob/35b4aa651a8ad7c1cfde431fa0e071e5c1a552b6/crates/tests-integration/tests/end_to_end_test.rs#L74)):*
> ```XML
>         }
>     }
>     assert_eq!(received_tx_hashes, expected_batched_tx_hashes);
> ```

Here we check the content, but I don't see where we check the propose'd block's ID.



_Code quote:_
```XML
   assert_eq!(received_tx_hashes, expected_batched_tx_hashes);
```

**Review by matan-starkware — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 4 unresolved discussions (waiting on @dan-starkware, @guy-starkware, and @Itay-Tsabary-Starkware)

___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 39 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK8IqA8GULjsFVSyTc:-OBK8IqA8GULjsFVSyTd:burgliz) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L39)):*
> ```XML
>         LISTEN_TO_BROADCAST_MESSAGES_TIMEOUT,
>         listen_to_broadcasted_messages(
>             mock_running_system.consensus_proposals_channels,
> ```

I want to point out that the interface here will change with Streaming. This is a flat `Receiver<ProposalPart>` where messages from all streams are mixed.

Streaming will have an interface of `Receiver<Receiver<ProposalPart>>` so that each distinct stream has a distinct channel to flow over. It might be best to change the context implementation now to that.



_Code quote:_
```XML
            mock_running_system.consensus_proposals_channels,
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 68 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK02SM3XWrcgtdT4iE:-OBK02SN6wZLqSr1mPJv:b9p5lta) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L68)):*
> ```XML
>         proposal_content_id: BlockHash(felt!(
>             "0x4597ceedbef644865917bf723184538ef70d43954d63f5b7d8cb9d1bd4c2c32"
>         )),
> ```

I prefer to minimize macro usage. 



_Suggestion:_
```XML
        proposal_content_id: BlockHash(Felt::from_hex_unchecked(
            "0x4597ceedbef644865917bf723184538ef70d43954d63f5b7d8cb9d1bd4c2c32",
        )),
```
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 98 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK6jcp9wUkbi2sPpn4:-OBK6jcp9wUkbi2sPpn5:b45h) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L98)):*
> ```XML
>             }
>         }
>     }
> ```





_Suggestion:_
```XML
    assert_eq!(
        broadcasted_messages_receiver.next().await.unwrap().0.unwrap(),
        ProposalPart::Init(expected_proposal_init)
    );
    loop {
        match broadcasted_messages_receiver.next().await.unwrap().0.unwrap() {
            ProposalPart::Init(init) => panic!("Unexpected init: {:?}", init),
            ProposalPart::Fin(proposal_fin) => {
                assert_eq!(proposal_fin, expected_proposal_fin);
                break;
            }
            ProposalPart::Transactions(transactions) => {
                received_tx_hashes.extend(
                    transactions
                        .transactions
                        .iter()
                        .map(|tx| tx.calculate_transaction_hash(&chain_id).unwrap()),
                );
            }
        }
    }
```

**Review by dan-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: 7 of 8 files reviewed, 4 unresolved discussions (waiting on @guy-starkware, @Itay-Tsabary-Starkware, and @matan-starkware)

___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 39 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK8IqA8GULjsFVSyTc:-OBKDFZZBF_U_2ezg9cB:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L39)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

I want to point out that the interface here will change with Streaming. This is a flat `Receiver<ProposalPart>` where messages from all streams are mixed.

Streaming will have an interface of `Receiver<Receiver<ProposalPart>>` so that each distinct stream has a distinct channel to flow over. It might be best to change the context implementation now to that.
</blockquote></details>

Done.
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 68 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK02SM3XWrcgtdT4iE:-OBKEC6G8G7joIObTMq8:b2uy1s0) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L68)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

I prefer to minimize macro usage.
</blockquote></details>

Why? This is a utility for testing.
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 98 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK6jcp9wUkbi2sPpn4:-OBKFt3x4lmKqxJOFddy:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L98)):*
> ```XML
>             }
>         }
>     }
> ```

Done.

**Review by matan-starkware — APPROVED** [review]
> Reviewed 7 of 7 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 2 unresolved discussions (waiting on @dan-starkware, @guy-starkware, and @Itay-Tsabary-Starkware)

___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 39 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK8IqA8GULjsFVSyTc:-OBUTGlpEsUK1sWm6ym_:bfcql4q) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L39)):*
<details><summary><i>Previously, dan-starkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I don't see any change, which is fine for this PR, just noting since you wrote "Done"
___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 68 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK02SM3XWrcgtdT4iE:-OBUTVTo2iyWYZXrzjMN:b-rs15wc) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L68)):*
<details><summary><i>Previously, dan-starkware wrote…</i></summary><blockquote>

Why? This is a utility for testing.
</blockquote></details>

macros are harder to work with in the IDE, you don't get nice type hints, and jumping to the definition doesn't bring you to the actual function used, still requires more sleuthing. They also make compile times slower.

I prefer functions when there is a 1 to 1 replacement.

**Review by dan-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: 5 of 13 files reviewed, 1 unresolved discussion (waiting on @guy-starkware, @Itay-Tsabary-Starkware, and @matan-starkware)

___
*[`crates/tests-integration/tests/end_to_end_test.rs` line 39 at r5](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-OBK8IqA8GULjsFVSyTc:-OBUnrvr6obW28654MQW:b-q4z4tg) ([raw file](https://github.com/starkware-libs/sequencer/blob/fff562bcb3cd5b25bffcdf8728ce3cee5f15657e/crates/tests-integration/tests/end_to_end_test.rs#L39)):*
<details><summary><i>Previously, matan-starkware wrote…</i></summary><blockquote>

I don't see any change, which is fine for this PR, just noting since you wrote "Done"
</blockquote></details>

I meant ACK

**Review by matan-starkware — APPROVED** [review]
> Reviewed 7 of 8 files at r8, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 1 unresolved discussion (waiting on @guy-starkware and @Itay-Tsabary-Starkware)

**Review by Itay-Tsabary-Starkware — APPROVED** [review]
> Reviewed 1 of 6 files at r5, 3 of 7 files at r7, 4 of 8 files at r8, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 1 unresolved discussion (waiting on @guy-starkware)

**Review by ShahakShama — APPROVED** [review]
:lgtm:

> Reviewed 1 of 6 files at r5, 2 of 7 files at r7, 1 of 8 files at r8, 10 of 10 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811#-:-OBVjLmeBMJii1iNyCwu:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @dan-starkware and @guy-starkware)

**Review by ShahakShama — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 2 unresolved discussions (waiting on @dan-starkware and @guy-starkware)

**Review by ShahakShama — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 2 unresolved discussions (waiting on @dan-starkware and @guy-starkware)

File: crates/tests_integration/tests/end_to_end_flow_test.rs [hunk 1]
```diff
@@ -34,60 +32,61 @@ async fn test_end_to_end(tx_generator: MultiAccountTransactionGenerator) {
...
+    let expected_proposal_init = ProposalInit {
+        height: BlockNumber(1),
+        round: 0,
+        valid_round: None,
+        proposer: ContractAddress::default(),
+    };
+    let expected_proposal_fin = ProposalFin {
+        proposal_content_id: BlockHash(Felt::from_hex_unchecked(
+            "0x4597ceedbef644865917bf723184538ef70d43954d63f5b7d8cb9d1bd4c2c32",
+        )),
```
[the remark this answers is not in the document]
Comment (dan-starkware) [reply crates/tests_integration/tests/end_to_end_flow_test.rs:68]: :thumbsup:

**Review by ShahakShama — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: all files reviewed, 2 unresolved discussions (waiting on @graphite-app[bot] and @guy-starkware)

**Review by yair-starkware — COMMENTED** [review]
> Dismissed @guy-starkware from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/1811)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @guy-starkware)

**Pull request merged**